gbm-sys/build.rs

```diff
@@ -1,33 +1,75 @@
 #[cfg(feature = "gen")]
 extern crate bindgen;
 
-#[cfg(feature = "gen")]
-use std::env;
-#[cfg(feature = "gen")]
-use std::path::Path;
+use std::{env, path::Path};
 
 #[cfg(not(feature = "gen"))]
-fn main() {}
+fn main() {
+    let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
+    let target_arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+
+    let bindings_file = Path::new("src")
+        .join("platforms")
+        .join(&target_os)
+        .join(&target_arch)
+        .join("gen.rs");
+
+    if bindings_file.is_file() {
+        println!(
+            "cargo:rustc-env=GBM_SYS_BINDINGS_PATH={}/{}",
+            target_os, target_arch
+        );
+    } else {
+        panic!(
+            "No prebuilt bindings for target OS `{}` and/or architecture `{}`. Try `gen` feature.",
+            target_os, target_arch
+        );
+    }
+}
 
 #[cfg(feature = "gen")]
-fn main()
-{
+fn main() {
     // Setup bindings builder
     let generated = bindgen::builder()
         .header("include/gbm.h")
-        .no_unstable_rust()
         .ctypes_prefix("libc")
-        .whitelisted_type(r"^gbm_.*$")
-        .whitelisted_function(r"^gbm_.*$")
+        .whitelist_type(r"^gbm_.*$")
+        .whitelist_function(r"^gbm_.*$")
         .constified_enum("gbm_bo_flags")
+        .constified_enum_module("gbm_bo_transfer_flags")
         .generate()
         .unwrap();
 
     println!("cargo:rerun-if-changed=include/gbm.h");
 
     // Generate the bindings
     let out_dir = env::var("OUT_DIR").unwrap();
-    let dest_path = Path::new(&out_dir).join("gen.rs");
+    let bind_name = "gen.rs";
+    let dest_path = Path::new(&out_dir).join(bind_name);
 
     generated.write_to_file(dest_path).unwrap();
+
+    #[cfg(feature = "update_bindings")]
+    {
+        use std::{fs, io::Write};
+
+        let bind_file = Path::new(&out_dir).join(bind_name);
+        let dest_dir = Path::new("src")
+            .join("platforms")
+            .join(env::var("CARGO_CFG_TARGET_OS").unwrap())
+            .join(env::var("CARGO_CFG_TARGET_ARCH").unwrap());
+        let dest_file = dest_dir.join(bind_name);
+
+        fs::create_dir_all(&dest_dir).unwrap();
+        fs::copy(&bind_file, &dest_file).unwrap();
+
+        if let Ok(github_env) = env::var("GITHUB_ENV") {
+            let mut env_file = fs::OpenOptions::new()
+                .create(true)
+                .append(true)
+                .open(github_env)
+                .unwrap();
```

Comment:
I think this is the default for [`File::create`](https://doc.rust-lang.org/std/fs/struct.File.html#method.create):

```suggestion
            let mut env_file = File::create(github_env).unwrap();
```

Reply:
We do not need `truncate(true)`, which is also default for `File::create`.

Reply:
In that case, please use `File::create`